Tensile/Source/client/include/DataInitialization.hpp

```diff
@@ -57,6 +57,16 @@ namespace Tensile
             Count
         };
 
+        enum class BoundCheckMode
+        {
+            Disable = 0,
+            NaN,
+            GuardPageFront,
+            GuardPageEnd
+        };
+
+        const int pageSize = 2 * 1024 * 1024;
```

Comment:
Are we sure the page size is always exactly 2MB?

Is there a way to query for the page size?

Reply:
@sdquiring No way to query the actual page size. Or I can add PageSize globalParameters config to let user set the pagesize? This may be better than hard-code design. Is that ok for you?

Reply:
No, I think this is fine.  If we need to have this configurable I think it should be through the `Hardware` class.